doc/modularity.rst

```diff
@@ -0,0 +1,94 @@
+..
+  Copyright (C) 2019 Red Hat, Inc.
+
+  This copyrighted material is made available to anyone wishing to use,
+  modify, copy, or redistribute it subject to the terms and conditions of
+  the GNU General Public License v.2, or (at your option) any later version.
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY expressed or implied, including the implied warranties of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+  Public License for more details.  You should have received a copy of the
+  GNU General Public License along with this program; if not, write to the
+  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+  02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+  source code or documentation are not subject to the GNU General Public
+  License and may only be used or replicated with the express permission of
+  Red Hat, Inc.
+
+
+############
+ Modularity
+############
+
+
+=============
+ Definitions
+=============
+
+modulemd
+    Every repository can contain "modules" metadata with modulemd documents.
+    These documents hold metadata about modules such as Name, Stream or list of packages.
+
+(non-modular) package
+    Package that doesn't belong to a module.
+
+modular package
+    Package that belongs to a module. It is listed in modulemd.
```

Comment:
Should we add additional details like: It is listed in modulemd metadata file (yaml) in section Artifacts.

Reply:
updated